### PR TITLE
test: test-erasure-eio.sh fix recovery testing and enable it

### PR DESCRIPTION
**Pull request by dzafman**
Fix and enable additional testing that wasn't possible without EC improvements.

### CODE REVIEW

**Comment by dachary** [comment]
jenkins test this please (background ceph-helper.sh race http://tracker.ceph.com/issues/18018)

File: src/test/erasure-code/test-erasure-eio.sh [hunk 1]
```diff
@@ -122,17 +122,19 @@ function rados_put_get() {
...
-        # check the object can still be retrieved, which implies
-        # recovery
+        # take out the last OSD used to store the object,
+        # bring it back, and check for clean PGs which means
+        # recovery didn't crash the primary.
         #
         local -a initial_osds=($(get_osds $poolname $objname))
         local last=$((${#initial_osds[@]} - 1))
+        # Kill OSD
+        kill_daemons $dir TERM osd.${initial_osds[$last]} >&2 < /dev/null || return 1
```
Comment (tchaikov) [inline src/test/erasure-code/test-erasure-eio.sh]: yeah, by bring an osd "out" does not move it out of the acting set.

Comment (tchaikov) [inline src/test/erasure-code/test-erasure-eio.sh]: but i am curious why we want to send an EOF to `kill_daemons()`.

Comment (dzafman) [inline src/test/erasure-code/test-erasure-eio.sh]: @tchaikov I copied how kill_daemons was called elsewhere.

Comment (dachary) [inline src/test/erasure-code/test-erasure-eio.sh]: @dzafman there is no need for it

**Pull request merged**